### PR TITLE
fix(toast): fix escape key behaviour when toast launched from dialog 

### PR DESCRIPTION
**Pull request by robinzigmond**
### Proposed behaviour

When a Toast is launched from a Dialog (or similar component using a Modal), after pressing Escape to close the Toast, a subsequent press of the Escape key should close the Dialog.

### Current behaviour

In the situation described above, pressing Escape does nothing once the Toast has been dismissed

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Cypress automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/awesome-banach-tip6mx?file=/src/index.js

### CODE REVIEW

File: src/hooks/__internal__/useModalEscape/useModalEscape.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useRef, useCallback } from "react";
+import ModalManager from "../../../components/modal/__internal__/modal-manager";
+
+const useModalEscape = (
+  open: boolean,
+  closeModal: (e: Event) => void,
+  modalRef: React.RefObject<HTMLElement>,
+  setTriggerRefocusFlag: (flag: boolean) => void | undefined
```
Comment (mkrds) [inline src/hooks/__internal__/useModalEscape/useModalEscape.ts:8]:
```suggestion
  setTriggerRefocusFlag?: (flag: boolean) => void | undefined
```

File: src/hooks/__internal__/useModalEscape/useModalEscape.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useRef, useCallback } from "react";
+import ModalManager from "../../../components/modal/__internal__/modal-manager";
+
+const useModalEscape = (
+  open: boolean,
+  closeModal: (e: Event) => void,
```
Comment (mkrds) [inline src/hooks/__internal__/useModalEscape/useModalEscape.ts:6]:
```suggestion
  closeModal: (e: KeyboardEvent) => void,
```

**Comment by mkrds** [comment]
I think that `useModalManager` would be a better name for the hook you've implemented. 
Also this bit of code `const isTopmost = ModalManager.isTopmost(ref.current);` which is used in both `toast.component` and `modal.component` could be abstracted to the `useModalManager` hook too. This way all the modal manager logic would be contained in `useModalManager`

**Review by mkrds — DISMISSED** [review]

File: src/components/confirm/confirm.spec.js [hunk 3]
```diff
@@ -111,17 +114,14 @@ describe("Confirm", () => {
...
-        domNode = wrapper.getDOMNode();
-        document.body.appendChild(domNode);
       });
 
       afterEach(() => {
-        document.body.removeChild(domNode);
         onCancelFn.mockReset();
       });
 
       it("should not close the modal if ESC key is pressed", () => {
```
Comment (grabkowski) [inline src/components/confirm/confirm.spec.js:123]: I think that we could change it to be what we are actually testing: `"onCancel" callback should not be called`

**Review by grabkowski — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**